### DISCUSSION TITLE
Is there a plan to support Avalonia

### DISCUSSION THREAD

**Discussion by oneo-me** [CommunityToolkit/Labs-Windows#534] [Ideas]
Here's an introduction to [Avalonia](https://github.com/AvaloniaUI/Avalonia):

> Develop Desktop, Embedded, Mobile and WebAssembly apps with C# and XAML. The most popular .NET Foundation community project.

![](https://user-images.githubusercontent.com/552074/235865745-2a8e7274-4f66-4f77-8f05-feeb76e7d478.png)

**Comment by Arlodotexe** [comment]
Thanks for your inquiry, glad to clear this up 🙂 

We don't have any plans to support Avalonia in the Windows Community Toolkit as we use .NET+WinUI, and WinUI isn't interchangeable with the XAML used by Avalonia, making it impractical to MultiTarget.

Our codebase is already WinUI, so there was very little code to change to target the platforms supported by Uno Platform. We more or less just added the MultiTarget TFMs and library packages and were good to go, minus a few small bumps that they were adept to help us work out. 

That said, if there were room in the CommunityToolkit org for Avalonia in a similar sense to the [Maui Community Toolkit](https://github.com/CommunityToolkit/Maui), @michael-hawker might know better what the barriers to entry are.
